### PR TITLE
QE: Add test to run cobbler sync via WebUI

### PR DESCRIPTION
**Pull request by ByteOtter**
## What does this PR change?

Add test to run cobbler sync via `Admin > Manager Configuration > Cobbler`.

## GUI diff

No difference.

- [x] **DONE**

## Documentation
- No documentation needed: only internal and user invisible changes

- [x] **DONE**

## Test coverage
- Cucumber tests were added

- [x] **DONE**

## Links

Fixes [#18806](https://github.com/SUSE/spacewalk/issues/18806)
Manager-4.3: https://github.com/SUSE/spacewalk/pull/19642
Manager-4.2: https://github.com/SUSE/spacewalk/pull/19643

- [x] **DONE**

## Changelogs

- [x] No changelog needed

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

### CODE REVIEW

File: testsuite/features/secondary/srv_cobbler_sync.feature [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+Feature: Run Cobbler Sync via WebUI
+
+  Scenario: Login as admin
+    Given I am authorized for the "Admin" section
+
+  Scenario: Sanity check for page existance
+    When I follow the left menu "Admin > Manager Configuration > Cobbler"
+    Then I should see a "Uyuni Configuration - Cobbler. " text
+    And I should see a "Setup your Uyuni Cobbler settings below. " text
```
Comment (nodeg) [inline testsuite/features/secondary/srv_cobbler_sync.feature:12]: You have to differentiate between SUMA and Uyuni here.

File: testsuite/features/secondary/srv_cobbler_sync.feature [hunk 2]
```diff
@@ -0,0 +1,23 @@
+# Copyright (c) 2022 SUSE LLC
+# Licensed under the terms of the MIT license.
+
+Feature: Run Cobbler Sync via WebUI
+
+  Scenario: Login as admin
+    Given I am authorized for the "Admin" section
+
+  Scenario: Sanity check for page existance
```
Comment (nodeg) [inline testsuite/features/secondary/srv_cobbler_sync.feature:9]: Please be more precice here. This Scenario description is to generic.

File: testsuite/features/secondary/srv_cobbler_sync.feature [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    When I follow the left menu "Admin > Manager Configuration > Cobbler"
+    Then I should see a "Uyuni Configuration - Cobbler. " text
+    And I should see a "Setup your Uyuni Cobbler settings below. " text
+    And I should see a "Cobbler sync is used to repair or rebuild the contents /srv/tftpboot or /srv/www/cobbler when manual modification of cobbler has occurred. " text
+    And I should see a "For more information refer to the 'cobbler' man page. " text
+    And I should see a "Run Cobbler Sync" text in the content area
+    And I should see a "Update" button
+
+  Scenario: Run Cobbler Sync via button and validate the task ran
```
Comment (nodeg) [inline testsuite/features/secondary/srv_cobbler_sync.feature:18]:
```suggestion
  Scenario: Run Cobbler Sync via button and validate the the result
```

**Review by nodeg — CHANGES_REQUESTED** [review]

File: testsuite/features/secondary/srv_cobbler_sync.feature (hunk 2 above)
Comment (Bischoff) [inline testsuite/features/secondary/srv_cobbler_sync.feature:9]:
```
Scenario: Cobbler Settings page exists
```

File: testsuite/features/secondary/srv_cobbler_sync.feature [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+    When I follow the left menu "Admin > Manager Configuration > Cobbler"
+    And I click on "Update"
+    Then I should see a "Cobbler Sync action was successfully executed. Look at /var/log/cobbler/*.log for more information" text
+    When I follow the left menu "Admin > Task Engine Status > Last Execution Times"
```
Comment (Bischoff) [inline testsuite/features/secondary/srv_cobbler_sync.feature:22]: In a separate scenario

File: testsuite/features/secondary/srv_cobbler_sync.feature [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+    Then I should see the correct timestamp for task "Cobbler Sync:"
```
Comment (Bischoff) [inline testsuite/features/secondary/srv_cobbler_sync.feature:23]: oh, cool, i did not notice that one

**Review by Bischoff — COMMENTED** [review]
see nitpicks, but other than that, looks very good to me

File: testsuite/features/secondary/srv_cobbler_sync.feature (hunk 1 above)
Comment (ByteOtter) [reply testsuite/features/secondary/srv_cobbler_sync.feature:12]: Oh right completely forgot I will change this for the ports

File: testsuite/features/secondary/srv_cobbler_sync.feature (hunk 2 above)
Comment (ByteOtter) [reply testsuite/features/secondary/srv_cobbler_sync.feature:9]: Right, I have changed it comitting now

File: testsuite/features/secondary/srv_cobbler_sync.feature (hunk 3 above)
Comment (ByteOtter) [reply testsuite/features/secondary/srv_cobbler_sync.feature:18]: changed it

File: testsuite/features/secondary/srv_cobbler_sync.feature (hunk 1 above)
Comment (ByteOtter) [reply testsuite/features/secondary/srv_cobbler_sync.feature:12]: So how would I circumvent that? not check for the text at all? 🤔

Comment (nodeg) [reply testsuite/features/secondary/srv_cobbler_sync.feature:12]: You can use tags for that, e.g. `@uyuni` and `@susemanager` but I am not sure if you can use it in scenarios directly or just directly above a scenario.

File: testsuite/features/secondary/srv_cobbler_sync.feature [hunk 6]
```diff
@@ -0,0 +1,27 @@
+# Copyright (c) 2022 SUSE LLC
+# Licensed under the terms of the MIT license.
+
+Feature: Run Cobbler Sync via WebUI
+
+  Scenario: Login as admin
+    Given I am authorized for the "Admin" section
+
+  Scenario: Cobbler Settings Page exists
```
Comment (nodeg) [inline testsuite/features/secondary/srv_cobbler_sync.feature:9]:
```suggestion
  Scenario: Check that the Cobbler Settings Page exists
```
Use complete sentences please :wink:

File: testsuite/features/secondary/srv_cobbler_sync.feature (hunk 1 above)
Comment (ByteOtter) [reply testsuite/features/secondary/srv_cobbler_sync.feature:12]: I have only seen them used directly above scenarios ... I changed the text in the SUMA Ports but I'll take a closer look when I get to the office

Comment (srbarrios) [reply testsuite/features/secondary/srv_cobbler_sync.feature:12]: Do you need to repeat the scenario twice, one with a susemanager tag, and another with a uyuni tag, and then the different text on it.
That's the simplest way, the complex would be to change the text inside the step definition, but I would prefer the simple approach, as we are already using it in multiple palces.

**Review by nodeg — APPROVED** [review]

**Pull request merged**